internal/launchr/types.go

```diff
@@ -153,3 +158,15 @@ type ManagedFS interface {
 	fs.FS
 	FS() fs.FS
 }
+
+// ProcessorDiscoveryPlugin is an interface for discovering processors.
+type ProcessorDiscoveryPlugin interface {
```

Comment:
It doesn't seem we need to define a new plugin type. The behavior you try to define must handled by `action.Manager`. You must define a new method there and register your processors on `OnAppInit`.

Move all related functionality there. Consider implementing something similar to `WithContainerRunEnvironmentConfig`.